rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 2*time.Minute, "Maximum time a query may take before being aborted.")
+)
+
+var queryTimeoutErr = fmt.Errorf("query timeout after %v", *queryTimeout)
```

Comment:
At some point in the future, we can also think about reifying queries: that is, instead of passing around a TimerGroup object for everything that happens in a query, we'd pass around a Query object which then would register itself centrally somehow and be accessible (list queries, cancel queries, etc.) through an API. The timers would then be subobjects of that Query object. But that's only sci-fi talk now.